mytoyota/models/vehicle.py

```diff
@@ -247,19 +248,27 @@ def lock_status(self) -> Optional[LockStatus]:
             self._endpoint_data["status"] if "status" in self._endpoint_data else None
         )
 
-    async def get_summary(
+    async def get_summary(  # pylint: disable=R0914, R0912, R0915
```

Comment:
The whole `get_summary` code is complex due to repetitive sorting operations, redundant initializations of the return list ret, and unnecessary checks for None in accumulation logic.

I think it can be imporved by removing the redundant initializations of ret outside of the conditional branches since it's re-initialized in each branch.
You can also consolidate the sorting of resp.payload.summary before the conditional branches to avoid sorting in each branch. And simplify the accumulation logic by initializing build_hdc and build_summary once at the start of the loop instead of checking for None every time.
The functions for collecting the statistical data within the `for summary...` loops should also be outsourced to independent functions.

Reply:
The logic is complicated. I have attempted to improve it(in changes shortly) and have added more comments.

build_hdc and build_summary can not be initialised once. For example, if you are building weekly summaries for multiple months you dont know where the week ends, it could be in the summary loop or the histogram loop. To deal with this I reset them to None each time we find a Sunday so we can restart building the next week.

Reply:
I'm just thinking out loud and need to give it some more thought and try it myself: But wouldn't it be possible to make the code more consistent and uniform if we built some helper functions for week, month and year?
These helper functions are given a start date and end date, which they use to generate the weeks, months and years within them with their respective start and end dates.
With these generated start and end dates of the respective time periods, it should then be possible to filter the summary data consistently, right?

Reply:
Something like this for example for weeks:

```python
from datetime import datetime, timedelta

def kalenderwochen_dict(startdatum, enddatum):
    start_date = datetime.strptime(startdatum, "%Y-%m-%d")
    end_date = datetime.strptime(enddatum, "%Y-%m-%d")

    kalenderwochen_dict = {}

    current_date = start_date
    while current_date <= end_date:
        kw_start = current_date - timedelta(days=current_date.weekday())
        kw_end = kw_start + timedelta(days=6)

        kalenderwochen_dict[f"{current_date.year}-{current_date.isocalendar()[1]}"] = {
            'startdatum': kw_start.strftime("%Y-%m-%d"),
            'enddatum': kw_end.strftime("%Y-%m-%d")
        }

        current_date += timedelta(weeks=1)

    return kalenderwochen_dict

# Beispielaufruf:
startdatum = "2023-01-01"
enddatum = "2023-12-31"
resultat = kalenderwochen_dict(startdatum, enddatum)
print(resultat)
```

Reply:
I did think about this, but didn't implement as you use different parts of the summary response depending on what has been requested.
- Daily: Simply pull out the data returned in the histograms, no maths required.
- Weekly: Add together the days that make up a week from the histograms.
- Monthly: Simply pull out the monthly summary supplied in the data, no maths required. NOTE: No need to look at the histograms.
- Yearly: Add together the monthly summaries that make up the year. NOTE: No need to look at the histograms.

Yes, it could all be done with the histograms but having been provided with the Monthly data I made use of it as opposed to doing a lot of extra calculations on the histogram data.

Reply:
Good point! You're right, we shouldn't do manual calculations where they're not absolutely necessary.
Your recent changes have also made the code much better structured and easier to understand.

Reply:
OK, after thinking about it again, the question has arisen as to what we want to return and what the user expects.
At the moment, we deliver all units for all full months of the start and end date that have already been pre-calculated by Toyota, especially for the monthly values, right?
As a user, however, I would expect that when a start and end date is specified, exactly this time span is used for an aggregation at monthly level, without the additional days before and after for the full months.